src/libs/plugin/plugin.c

```diff
@@ -152,6 +152,32 @@ void * elektraPluginGetData (Plugin * plugin)
 	return plugin->data;
 }
 
+/**
+ * @brief get the KDB instance this plugin is loaded into
+ *
+ * @param plugin a pointer to the plugin
+ * @return KDB
+ */
+KDB * elektraPluginGetKdb (Plugin * plugin)
+{
+	if (plugin->global == NULL)
+	{
+		return NULL;
+	}
+
+	Key * kdbKey = ksLookupByName (plugin->global, "system:/elektra/kdb", 0);
```

Comment:
AFAICT this is key is always present, so I guess it is okay to rely on this?

Reply:
Once we document that the key must be there, we can assume it. (And it would be a bug if it isn't.)

Reply:
It's created in `kdbNew`, i.e. almost immediately after the `struct _KDB` memory is allocated. 

https://github.com/ElektraInitiative/libelektra/blob/0082b7012d7a974cee5ee82098fd0dcb38274828/src/libs/elektra/kdb.c#L295-L296

IIRC I added this key, for a purpose exactly like this one. But I didn't create a function in the plugin API, because I wasn't sure whether letting plugins access the whole `KDB` instance is a good idea.

That's also why `elektraPluginFromMountpoint` takes a `Plugin *` and retrieves the necessary data internally. If we keep the `elektraPluginGetKdb` function, we should definitely ensure that plugins cannot call any of the `kdb*` functions on the `KDB *` the receive. For example, via a `callInProgress` flag in `struct _KDB`.

Reply:
Yeah, I agree it's better to create a function that takes a `Plugin *` and returns the `ChangeTrackingContext *`. It's also cleaner for the tests.